### ISSUE TITLE
[BUG] You can't migrate a VM off of a host that is unavailable

### ISSUE THREAD

**Issue by noahgildersleeve** [harvester/harvester#983]
**Describe the bug**

You can't migrate a VM off of a host that is unavailable
**To Reproduce**
1. Create a new VM on a host in a cluster with at least 4 nodes
2. Turn off that host
3. Try to migrate the VM to a new host
**Expected behavior**

The VM should be able to be migrated
**Support bundle**

[harvester-supportbundle_c77a57eb-bc3d-4ed3-9037-6f9da0c0e2f6_2021-06-28T23-39-53Z.zip](https://github.com/harvester/harvester/files/6729711/harvester-supportbundle_c77a57eb-bc3d-4ed3-9037-6f9da0c0e2f6_2021-06-28T23-39-53Z.zip)

**Environment:**
 - Harvester ISO version: master-158a35d3-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): bare metal on DL160

**Additional context**
It gets stuck on migrating. 
![Greenshot 2021-06-28 16 46 26](https://user-images.githubusercontent.com/83787952/123716828-6a658500-d830-11eb-861d-636d92311e56.png)

**Comment by guangyee** [comment]
We basically cannot do anything to the VM right now if the host where it is running suddenly goes offline.

**Comment by noahgildersleeve** [comment]
@guangbochen This was marked as blocked by https://github.com/harvester/harvester/issues/982 but was moved into ready for testing? Is it not blocked anymore?

**Comment by noahgildersleeve** [comment]
Tested in master-a412145f-head.
1. Turned off host that had VMs on it
2. Waited for host to show as powered off in ILO
3. Tried to migrate VM to new host
I got an error that said that the VM wasn't in ready status and a 500 error in the console.
![Greenshot 2021-11-08 11 55 34](https://user-images.githubusercontent.com/83787952/140809207-a8fddfc9-0db2-41d4-a4c2-ca75e6ad7fde.png)
![Greenshot 2021-11-08 11 55 39](https://user-images.githubusercontent.com/83787952/140809210-0997c269-629d-40bc-a5d4-71173e8359f5.png)
![Greenshot 2021-11-08 11 56 04](https://user-images.githubusercontent.com/83787952/140809212-8d351afe-f436-4254-ba56-f46a7834e7a5.png)
![Greenshot 2021-11-08 11 56 36](https://user-images.githubusercontent.com/83787952/140809213-86f036fc-fa8a-4f95-8b94-f2df10c643cf.png)

**Comment by FrankYang0529** [comment]
@noahgildersleeve, I think it is expected behavior. We can't "Live Migrate" a VM from a shutdown node, so we show the error message to users.

We already have another PR to handle VMs on a shutdown node. It needs users to set `vm-force-deletion-policy`. https://github.com/harvester/harvester/pull/1502

**Comment by noahgildersleeve** [comment]
Closing as working as designed.
